src/ocean/THCM.C

```diff
@@ -3018,3 +2979,161 @@ Teuchos::RCP<const Epetra_MultiVector> THCM::getNullSpace()
     return nullSpace;
 }
 
+//=============================================================================
+Teuchos::ParameterList&
+THCM::setDefaultInitParameters(Teuchos::ParameterList& params)
+{
+    params.get("Problem Description","Unnamed");
+
+    params.get("Grid Stretching qz", 1.0);
+    params.get("Topography", 1);
+    params.get("Flat Bottom", false);
+
+    params.get("Read Land Mask", false); //== false in experiment0
+    params.get("Land Mask","no_mask_specified");
+
+    params.get("Coriolis Force", 1);
+    params.get("Forcing Type", 0);
+
+    params.get("Read Salinity Perturbation Mask",false);
+    params.get("Salinity Perturbation Mask", "no_mask_specified");
+
+    params.get("Wind Forcing Data", "wind/trtau.dat");
+    params.get("Temperature Forcing Data", "levitus/new/t00an1");
+    params.get("Salinity Forcing Data", "levitus/new/s00an1");
+
+    params.get("Time Dependent Forcing", false);
+
+    THCM::setDefaultParameters(params);
+
+    int N = params.get<int>("Global Grid-Size n");
+    int M = params.get<int>("Global Grid-Size m");
+
+    params.get("Integral row coordinate i", N-1);
+    params.get("Integral row coordinate j", M-1);
```

Comment:
@Sbte Only problem with the idea of using validateParametersAndSetDefaults are these two lines, since their defaults depends on the values for the `Global Grid-Size`, which works in the current version, but can't really be done using set, I think?

The only option would be to set the default value equal to the default `Global Grid-Size`, but then any code with a different grid size needs to also explicitly set these.

Reply:
They actually don't seem to be explicitly set anywhere, except for `test_integrals` (where they're unused) and `test/global/ocean_params.xml` which doesn't appear to be used anywhere either...